src/datasets/misc/wine.jl

```diff
@@ -0,0 +1,86 @@
+export Wine
+
+"""
+    Wine(; as_df = true, dir = nothing)
+
+The UCI Wine dataset.
```

Comment:
can you add a reference to the original data source or paper?

Reply:
Done!

Also, I take advantage of this to say that I intentionally avoided to insert the getproperty method as I've seen it is deprecated.  If it should be inserted for compatibility just let me know.

Anyway, great job! I saved a lot of time yesterday!

Reply:
No need for the deprecated methods, I actually they have been around for long time, we should fist remove them